plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,222 @@
+      - Dynamic query updates the search results before execution (useful for scheduled jobs)
+    choices:
+      - static_query
+      - dynamic_query
+    default: static_query
+    type: str
+  randomized_ordering:
+    description:
+      - Whether to order the selected hosts randomly
+    type: bool
+  execution_timeout_interval:
+    description:
+      - Override the timeout interval from the template for this invocation only
+    type: int
+  ssh:
+    description:
+      - ssh related options
+    type: dict
+    suboptions:
+      effective_user:
+        description:
+          - What user should be used to run the script (using sudo-like mechanisms)
+          - Defaults to a template parameter or global setting
+        type: str
+  feature:
+    description:
+      - Remote execution feature label that should be triggered, job template assigned to this feature will be used
+    type: str
+  command:
+    description:
+      - Command to be executed on host. Required for command templates
+    type: str
+  inputs:
+    description:
+      - Inputs to use
+    type: dict
+  recurrence:
+    description:
+      - Schedule a recurring job
+    type: dict
+    suboptions:
+      cron_line:
+        description:
+          - How often the job should occur, in the cron format
+        type: str
+      max_iteration:
+        description:
+          - Repeat a maximum of N times
+        type: int
+      end_time:
+        description:
+          - Perform no more executions after this time
+        type: str
+  scheduling:
+    description:
+      - Schedule the job to start at a later time
+    type: dict
+    suboptions:
+      start_at:
+        description:
+          - Schedule the job for a future time
+        type: str
+      start_before:
+        description:
+          - Indicates that the action should be cancelled if it cannot be started before this time.
+        type: str
+  concurrency_control:
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template: "Run Command - SSH Default"
+    ssh:
+      effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark: 'active'
+    command: 'pwd'
+    job_template: "Run Command - Ansible Default"
+    recurrence:
+      cron_line: "30 2 * * *"
+    concurrency_control:
+      concurrency_level: 2
+'''
```

Comment:
I think, this module has a lot of complexity. Would you mind adding more examples, using `feature` and `inputs`?

Reply:
I decided to remove feature for now, basically it is an another way of selecting a template, the available features can be listed under Administer > Remote Execution Features in the UI. There's just a few of them and none binds to the two templates the module currently supports.
I'll add inputs example based on the result of discussion https://github.com/theforeman/foreman-ansible-modules/pull/864#discussion_r491867661